app/components/file/browse/node-file-list.component.ts

```diff
@@ -43,63 +38,73 @@ export class NodeFileListComponent implements OnInit, OnChanges {
-    public notFound: boolean;
+    @ViewChild(TreeViewDisplayComponent)
+    public treeDisplay: TreeViewDisplayComponent;
+    public moreFileMap: StringMap<boolean> = {};
+    public status: BehaviorSubject<LoadingStatus> = new BehaviorSubject(LoadingStatus.Loading);
+    public error: BehaviorSubject<ServerError> = new BehaviorSubject(null);
 
-    constructor(private fileService: FileService) {
-        this.notFound = false;
-        this.data = this.fileService.listFromComputeNode(null, null, true, {});
-        this.status = this.data.status;
-    }
+    private _fileProxyMap: StringMap<RxListProxy<NodeFileListParams, File>> = {};
 
-    public ngOnInit() {
-        return;
-   }
+    constructor(private treeComponentService: TreeComponentService, private fileService: FileService) { }
 
     public ngOnChanges(inputs) {
         if (inputs.poolId || inputs.nodeId || inputs.folder || inputs.filter) {
+            this._initProxyMap(inputs);
             this.refresh();
         }
     }
 
     @autobind()
     public refresh(): Observable<any> {
-        if (!(this.poolId && this.nodeId)) {
-            return;
+        if (this.poolId && this.nodeId) {
+            // filter is changed in two different situation (search field of node file list and file nav list)
+            const filterProp = ((this.filter.properties && this.filter.properties.length > 0) ?
+                                        this.filter.properties[0] : this.filter) as Property;
+            const quickSearch = filterProp && filterProp.value;
+            const loadPath = [this.folder, quickSearch].filter(x => Boolean(x)).join("/");
+            if (this.treeDisplay) {
+                return this.treeDisplay.initNodes(loadPath, true);
+            }
         }
-        let options = {};
-        const filter = this._buildFilter();
-        if (!filter.isEmpty()) {
-            options = {
-                filter: filter.toOData(),
-            };
-        }
-        this.data.updateParams({ poolId: this.poolId, nodeId: this.nodeId });
-        this.data.setOptions(options); // This clears the previous list objects
-        this.notFound = false;
-        return this.data.fetchNext(true);
-    }
-
-    @autobind()
-    public loadMore(): Observable<any> {
-        return this.data.fetchNext();
+        return Observable.of(true);
     }
 
     public get baseUrl() {
         return ["/pools", this.poolId, "nodes", this.nodeId];
     }
 
-    private _buildFilter() {
-        const filter: Property = this.filter as Property;
-        const quickSearch = filter && filter.value;
+    @autobind()
+    public loadPath(path: string, refresh: boolean = false): Observable<List<File>> {
+        if (!(path in this._fileProxyMap)) {
+            const options = buildTreeRootFilter(path);
+            const poolId = this.poolId;
+            const nodeId = this.nodeId;
+            this._fileProxyMap[path] = this.fileService.listFromComputeNode(poolId, nodeId, false, options);
```

Comment:
you are going to need to call dispose on those listproxy or there is going to be a memroy leak. Need to take care of that when you clear the map too becaues inputs changes